### ISSUE TITLE
Continous game_sa.dll crash with latest MTA version 1.5.7-release-20658.0.000

### ISSUE THREAD

**Issue by haron4igg** [multitheftauto/mtasa-blue#1695]
Player reports  continous crash with latest MTA version 1.5.7-release-20658.0.000
Before latest update he never experienced this one.

-- Unhandled exception -- 

Version = 1.5.7-release-20658.0.000
Time = Mon Oct  5 15:47:28 2020
Module = D:\My\Games\MTA\game_sa.dll
Code = 0xC0000409
Offset = 0x0000EFFA

EAX=00000000  EBX=0028EDE8  ECX=58321558  EDX=00000000  ESI=0028EDE8
EDI=3B2610A8  EBP=0028ED58  ESP=0028ED30  EIP=6FAAEFFA  FLG=00200246
CS=0023   DS=002B  SS=002B  ES=002B   FS=0053  GS=002B

[client_1.5.7-release-20658.0.000_gamesa_0000effa_409_DMGMM_C269DE53_55F3_04B_D9B70_20201005_1547.rsa.zip](https://github.com/multitheftauto/mtasa-blue/files/5326799/client_1.5.7-release-20658.0.000_gamesa_0000effa_409_DMGMM_C269DE53_55F3_04B_D9B70_20201005_1547.rsa.zip)

=============================

MTADiag: https://pastebin.mtasa.com/667249675

=============================

Issue appears randomly, during connection, or during gameplay.

**Comment by ghost** [comment]
<details>
  <summary>Call stack screenshot from your crash dump</summary>

![image](https://user-images.githubusercontent.com/10183157/95077417-e5c85900-072c-11eb-82c3-c155823ed5c5.png)
</details>
<details>
  <summary>Call stack screenshot from a different dump file but the same crash</summary>

![image](https://user-images.githubusercontent.com/10183157/95076581-7ef67000-072b-11eb-9cd4-3baf0afe33b4.png)
</details>

The crash happened when replacing an object model via `engineReplaceModel` function. This looks like a corrupted model issue. Have you recently added any new object models? The name of the custom object model frame: `prison_fence_large_noba`

**Comment by haron4igg** [comment]
I've been observing this kind of model corruption crash when player reconnects same server many times. like after 3-4 reconnects without MTA restart it will crash due to model corruption. 

There is one more thing we observed with this player right now. 
This crash starts happening after another one:

** -- Unhandled exception -- **

Version = 1.5.7-release-20658.0.000
Time = Mon Oct  5 15:44:31 2020
Module = D:\My\Games\MTA\MTA Laucher\MTA\libcef.dll
Code = 0x80000003
Offset = 0x01EA8E5E

EAX=00000000  EBX=0028F3DC  ECX=00000000  EDX=8781D9D8  ESI=0028F81C
EDI=0028F824  EBP=0028F814  ESP=0028F3B8  EIP=82E68E5E  FLG=00200246
CS=0023   DS=002B  SS=002B  ES=002B   FS=0053  GS=002B

** -- End of unhandled exception -- **

[client_1.5.7-release-20658.0.000_libcef_01ea8e5e_3_DMGMM_C269DE53_55F3_432_D9B70_20201005_1605.rsa.zip](https://github.com/multitheftauto/mtasa-blue/files/5327191/client_1.5.7-release-20658.0.000_libcef_01ea8e5e_3_DMGMM_C269DE53_55F3_432_D9B70_20201005_1605.rsa.zip)


So. We have setup of 2 servers with exacltly same Models Pack (same git revision). 
Server A was updated 2 days ago.
Server B was updated this morning (only game logic, translations, a bit of js logic and some minor mapping movements since version A). 

When player join server A he have no issues neither with CEF, neither with those game_sa.dll crash.
If player join server B, everything runs OK until he access CEF menu. Once he open CEF, he get this crash from libcef.
After he restarts MTA, when he is trying to join server B - he get this game_sa.dll crash.
If he trying to join A - everytinhg is ok. Then if he join to B - everting is ok, until CEF menu is accessed again. 
Even if i stop all cef resources, he will anyway keep getting this game_sa.dll crash, until he join server A.

1389 object id, the model was replaced like 6 years ago. Anybody who experince this game_sa.dll crash have it on this model. But i been trying already to remove it, replace it, etc. Crash just goes to next one in the replace list :D

So looks like it is caused by some changes in CEF ui...

**Comment by ghost** [comment]
> 1389 object id, the model was replaced like 6 years ago. Anybody who experince this game_sa.dll crash have it on this model. But i been trying already to remove it, replace it, etc. Crash just goes to next one in the replace list :D

Are you able to reproduce the crash in the game_sa.dll module? If possible, send me the model, I'll check if I can force it to crash. About the CEF crash, it can happen if your MTA process runs out of memory.

**Comment by haron4igg** [comment]
[objects.zip](https://github.com/multitheftauto/mtasa-blue/files/5327360/objects.zip)

100% the model isn't an issue here. Because in past i've been trying to fix it, was trying to remove it, corrupted model crash just flows to the next model in the list:

![image](https://user-images.githubusercontent.com/4466653/95080705-2b7f2480-0721-11eb-9edc-d4e194be0531.png)

i would say it is something related to memory-leaks in gta engine, because in normal situation it happens only after 3-4 reconnects during same mta session.

**Comment by ghost** [comment]
Is the CEF crash problem related to #1180? Moreover, is it possible for you to temporarily disable CEF on your server and see if that helps? CEF does eat a lot of ram depending on what you're rendering. A few months ago a server contacted me about the same issue that their server was randomly crashing in libcef module. It's probably better to unload modules that are not in being used to avoid running out of memory address space.

**Comment by haron4igg** [comment]
Ok we found issue here, there was intence, almost recurive operation in JS and for this user, memory wasn't freeing fast anough. But... Now issue fixed, but after continuos usage of same  CEF ui, like ~10 minutes, he still gets cef related crash and not able to connect anymore with game_sa.dll crash to any server, even where CEF ui is not released yet. Only full PC restart helps.

Just in case, we profiled CEF ui on his machine and i haven't found any js leaks or intense operations... so its still a mistery.
